pkg/utils/procfs/procfs.go

```diff
@@ -0,0 +1,247 @@
+	// NonvoluntaryCtxtSwitches int
+}
+
+// ReadStatus reads /proc/[pid]/status or /proc/[pid]/task/[tid]/status files
+func ReadStatus(filePath string) (ProcStatus, error) {
+	statusBytes, err := os.ReadFile(filePath)
+	if err != nil {
+		logger.Debugw("opening status file", "error", err)
+		return ProcStatus{}, err
+	}
+
+	statusMap := map[string]string{}
+	status := strings.Split(string(statusBytes), "\n")
+
+	for _, line := range status {
+		if len(line) < 2 {
+			continue
+		}
+		normal := strings.Replace(line, "kB", " ", -1)
+		fields := strings.Fields(normal)
+		statusMap[strings.Trim(fields[0], ":")] = strings.Join(fields[1:], ",")
+	}
+
+	procStatus := ProcStatus{
+		Name:   statusMap["Name"],
+		State:  statusMap["State"],
+		Groups: statusMap["Groups"],
+	}
+
+	_, _ = fmt.Sscanf(statusMap["Umask"], "%d", &procStatus.Umask)
+	_, _ = fmt.Sscanf(statusMap["Tgid"], "%d", &procStatus.Tgid)
+	_, _ = fmt.Sscanf(statusMap["Ngid"], "%d", &procStatus.Ngid)
+	_, _ = fmt.Sscanf(statusMap["Pid"], "%d", &procStatus.Pid)
+	_, _ = fmt.Sscanf(statusMap["PPid"], "%d", &procStatus.PPid)
+	_, _ = fmt.Sscanf(statusMap["TracerPid"], "%d", &procStatus.TracerPid)
+	_, _ = fmt.Sscanf(statusMap["Uid"], "%d\t%d\t%d\t%d", &procStatus.Uid)
+	_, _ = fmt.Sscanf(statusMap["Gid"], "%d\t%d\t%d\t%d", &procStatus.Gid)
+	_, _ = fmt.Sscanf(statusMap["FDSize"], "%d", &procStatus.FDSize)
+	_, _ = fmt.Sscanf(statusMap["NStgid"], "%d", &procStatus.NStgid)
+	_, _ = fmt.Sscanf(statusMap["NSpid"], "%d", &procStatus.NSpid)
+	_, _ = fmt.Sscanf(statusMap["NSpgid"], "%d", &procStatus.NSpgid)
+	_, _ = fmt.Sscanf(statusMap["NSsid"], "%d", &procStatus.NSsid)
+	_, _ = fmt.Sscanf(statusMap["VmPeak"], "%d", &procStatus.VmPeak)
+	_, _ = fmt.Sscanf(statusMap["VmSize"], "%d", &procStatus.VmSize)
+	_, _ = fmt.Sscanf(statusMap["VmLck"], "%d", &procStatus.VmLck)
+	_, _ = fmt.Sscanf(statusMap["VmPin"], "%d", &procStatus.VmPin)
+	_, _ = fmt.Sscanf(statusMap["VmHWM"], "%d", &procStatus.VmHWM)
+	_, _ = fmt.Sscanf(statusMap["VmRSS"], "%d", &procStatus.VmRSS)
+	_, _ = fmt.Sscanf(statusMap["RssAnon"], "%d", &procStatus.RssAnon)
+	_, _ = fmt.Sscanf(statusMap["RssFile"], "%d", &procStatus.RssFile)
+	_, _ = fmt.Sscanf(statusMap["RssShmem"], "%d", &procStatus.RssShmem)
+	_, _ = fmt.Sscanf(statusMap["VmData"], "%d", &procStatus.VmData)
+	_, _ = fmt.Sscanf(statusMap["VmStk"], "%d", &procStatus.VmStk)
+	_, _ = fmt.Sscanf(statusMap["VmExe"], "%d", &procStatus.VmExe)
+	_, _ = fmt.Sscanf(statusMap["VmLib"], "%d", &procStatus.VmLib)
+	_, _ = fmt.Sscanf(statusMap["VmPTE"], "%d", &procStatus.VmPTE)
+	_, _ = fmt.Sscanf(statusMap["VmSwap"], "%d", &procStatus.VmSwap)
+	_, _ = fmt.Sscanf(statusMap["HugetlbPages"], "%d", &procStatus.HugetlbPages)
+	_, _ = fmt.Sscanf(statusMap["Threads"], "%d", &procStatus.Threads)
+
+	return procStatus, nil
+}
+
+// ReadStat reads /proc/[pid]/sat or /proc/[pid]/task/[tid]stat files
+func ReadStat(filePath string) (ProcStat, error) {
+	statBytes, err := os.ReadFile(filePath)
```

Comment:
Should we request for some capability to do that?

Reply:
I am 100% sure, but I don't think that you need any capabilities to access these files.
In the end, the `ps` command for example is able to do it without any special capabilities or `suid` flag.

Reply:
I believe I placed it inside initBPF (for some other reasoning) and the capabilities there weren't dropped yet (this is why there isn't anything raising capabilities for this call).

Reply:
Solved. Its together with the other initialization functions in the singleton now. And with correct capabilities.